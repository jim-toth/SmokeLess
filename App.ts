import Sentry from 'sentry-expo';

// TODO -> Use environment vars, see: https://docs.expo.io/versions/v32.0.0/guides/using-sentry/#__next
Sentry.config('https://92e853601ecc4e2abbadb2d8732b85a3@sentry.io/1532012').install();

import App from './src/App';
export default App;
