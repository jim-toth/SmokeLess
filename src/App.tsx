import * as React from 'react';
import { View } from 'react-native';
import { Sentry } from 'react-native-sentry';
import Config from 'react-native-config';

import createAppContainer from './navigation/AppContainer';
import { fetchWelcomeCompleted } from './db/SettingsRepository';
import SplashLoader from './components/atoms/SplashLoader';

import { AppStyle } from './Styles';

Sentry.config(Config.SENTRY_URL).install();

interface IAppState {
  isLoadingComplete: boolean;
  isWelcomeComplete: boolean;
}

export default class App extends React.Component<IAppState> {
  state = {
    isLoadingComplete: false,
    isWelcomeComplete: false
  };

  render() {
    if (!this.state.isLoadingComplete) {
      return (
        <SplashLoader
          startAsync={this._loadResourcesAsync}
          onError={this._handleLoadingError}
          onFinish={this._handleFinishLoading}
        />
      );
    } else {
      const AppContainer = createAppContainer(this.state.isWelcomeComplete);
      return (
        <View style={AppStyle.container}>
          {/* {Platform.OS === 'ios' && <StatusBar barStyle="default" />} */}
          <AppContainer />
        </View>
      );
    }
  }

  _loadResourcesAsync = async () : Promise<void> => {
    return Promise.all([
      // Ionicons.loadFont(),
      // Font.loadAsync({
      //   'saira': require('../assets/fonts/Saira-Regular.ttf'),
      //   'saira-bold': require('../assets/fonts/Saira-Bold.ttf'),
      // }),
    ]).then(() => undefined);
  };

  _handleLoadingError = (error:any) => {
    // Reports to Sentry
    console.warn(error);
  };

  _handleFinishLoading = async () => {
    const isWelcomeComplete = await fetchWelcomeCompleted();
    this.setState({ isLoadingComplete: true, isWelcomeComplete });
  };
}
