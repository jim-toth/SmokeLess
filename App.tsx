// TODO -> set up Sentry

import App from './src/App';
export default App;
