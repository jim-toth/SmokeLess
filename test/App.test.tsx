import 'react-native';
import * as React from 'react';
import App from '../src/App';
import { shallow } from 'enzyme';

describe('<App />', () => {
  it('renders loading screen', () => {
    const tree = shallow(<App />);
    expect(tree).toMatchSnapshot();
  });

  it('renders without loading screen', () => {
    const tree = shallow(<App skipLoadingScreen />);
    expect(tree).toMatchSnapshot();
  });
});
