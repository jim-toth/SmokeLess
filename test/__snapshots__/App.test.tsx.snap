// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders loading screen 1`] = `
<AppLoading
  onError={[Function]}
  onFinish={[Function]}
  startAsync={[Function]}
/>
`;

exports[`<App /> renders without loading screen 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <NavigationContainer />
</Component>
`;
